text/trace/0136-error_flagging.md

```diff
@@ -0,0 +1,89 @@
+
+While generic instrumentation can only provide a generic schema, end users are capable of making subjective decisions about their systems. And, as the end user, they should get to have the final call in what constitutes an error. In order to accomplish this, there must be a way to differentiate between errors flagged by instrumentation, and errors flagged by the end user.
+
+## Explanation
+
+The following changes add several missing features required for proper error reporting, and are completely backwards compatible with OpenTelemetry today.
+
+### Status Codes
+
+Currently, OpenTelemetry does not have a use case for differentiating between different types of errors. However, this use case may appear in the future. For now, we would like to reduce the number of status codes, and then add them back in as the need becomes clear. We would also like to differentiate between status codes which have not been
+set, and an explicit OK status set by an end user.
+
+* `UNSET` is the default status code.
+* `ERROR` represents all error types.
+* `OK` represents a span which has been explicitly marked as being free of errors, and should not be counted against an error budget. Note that only end users should set this status. Instead, instrumentation should leave the status as `UNSET` for normal operations.
+
+### `Status Source`
+
+A new Status Source field identifies the origin of the status code on the span. This is important, as statuses set by application developers and operators have been confirmed by the end user to be correct to the particular situation. Statuses set by instrumentation, on the other hand, are only following a generic schema.
+
+* `INSTRUMENTATION` is the default source. This is used for instrumentation contained within shared code, such as OSS libraries and frameworks. All instrumentation plugins shipped with OpenTelemetry use this status code.
+* `USER` identifies statuses set by the end user, either in application code or the collector.
+
+Analysis tools MAY disregard status codes, in favor of their own approach to error analysis. However, it is strongly suggested that analysis tools SHOULD pay attention to the status codes when set by `USER`, as it is a communication from the end-user and contains valuable information.
+
+### Status Mapping Schema
+
+As part of the specification, OpenTelemetry provides a canonical mapping of semantic conventions to status codes. This removes any ambiguity as to what OpenTelemetry ships with out of the box.
```

Comment:
I don't really understand this sentence.
Does this mean the semantic conventions will be expanded to provide a mapping of a library's/framework's return status to whether it should be `ERROR` or `OK` or something else? Can we provide a reasonable mapping anywhere other than for HTTP? Our semantic conventions target generic _concepts_ like DB calls, RPC or messaging but usually not concrete libraries (e.g. MongoDB return code X). Will we add something like that?

Reply:
Yes, I would like to improve these conventions over time, to ensure that instrumentation is normalized across languages and. For example, we do not want to label a MongoDB code an error in one client, but not an error in another. We should do our best to record in the spec what counts as an error in the instrumentation we provide, when those errors relate to something cross-langauge, like a database protocol. 

It's also not clear to me if "generic" concepts like DB will be sufficient in the long run. SQL, Mongo, and Redis are all very different protocols. But even if the DB conventions remains generic, we still need to describe how to map each protocol to this generic convention, to ensure the data that we emit is consistent. I would like to bring in experts from these different databases to help with this.

Reply:
I think we can provide reasonable mappings for databases, though the conventions will need to include special cases (but really, have you seen the db semantic conventions? It feels like it's mostly special cases 😄). We can probably provide something pretty good for messaging RPC too.  

Also, I'd bet we can make a pretty good go at a general convention as well.  If the operation results in an exception or a panic, it's probably an error (is that controversial?).

Reply:
Great minds think alike :)

I added a clarification to this section.

Reply:
I see! That sounds reasonable to me.

@tedsuo I think you forgot to push the clarification but once that's done we're good to go :shipit:

Reply:
@arminru derp! Pushed. :)